lib/Graph/Nodes.cpp

```diff
@@ -228,7 +234,7 @@ static void verifyTanh(NodeValue src, NodeValue dest) {
 }
 
 static void verifySoftMax(NodeValue src, NodeValue dest) {
-  checkSameType(src, dest);
+  checkSimilarType(src, dest);
```

Comment:
I think that isSimilar is not a very good name. Why not just compare the element type? It will work for integer and floats.

Reply:
I agree, the name is unclear :-) What I want to compare is not only the element type, but also the shape too. In other words, everything should be compared, except the quantization information (scale / offset)

Reply:
The shape is already checked in the next line (which could just be replaced with `checkSameShape()` to make that clear).

So you could just replace this `checkSimilarType()` with:
```
assert(src.getElementType() == dest.getElementType());
```

Reply:
@nadavrot @jfix71 OK, I removed 'isSimilar', and like for some other nodes, I just check the equality for the shape like for arithmetic operation.